src/vm/jitinterfacegen.cpp

```diff
@@ -224,10 +224,10 @@ void InitJITHelpers1()
         if (GCHeap::UseAllocationContexts())
         {
 #ifdef FEATURE_IMPLICIT_TLS
-            SetJitHelperFunction(CORINFO_HELP_NEWSFAST, JIT_NewS_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWSFAST_ALIGN8, JIT_NewS_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWARR_1_VC, JIT_NewArr1VC_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWARR_1_OBJ, JIT_NewArr1OBJ_MP_FastPortable);
+            SetJitHelperFunction(CORINFO_HELP_NEWSFAST, reinterpret_cast<void*>(JIT_NewS_MP_FastPortable));
```

Comment:
It is a good guideline for nice pure C++ code.

However, CoreCLR is not always a nice pure C++ code by design. There is a bunch of low-level code that needs interoperate with details of the underlying platform ABI. This place is one of those cases. We always try to hide as much of the repetitive boiler plate code into the macros to make the code readable (e.g. check `HELPER_METHOD_FRAME_BEGIN/HELPER_METHOD_FRAME_END` macros - it is one of the more extreme cases).